gatsby-node.js

```diff
@@ -34,7 +34,7 @@ exports.createPages = ({ graphql, boundActionCreators }) => {
           path: edge.node.slug, // required
           component: blogPost,
           context: {
-            slug: edge.node.slug,
+            slug: { eq: edge.node.slug },
```

Comment:
It's a constraints of the runtime codified in the compiler. I agree the current way here isn't gonna work it's just to get it to compiler right now.  I have a few ideas to try still but they are all a bit hacky. This part of the compiler may also be pluggable.